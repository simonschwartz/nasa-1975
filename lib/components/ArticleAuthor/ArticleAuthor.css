.NASA1975-article-author {
  list-style-type: none;
  padding: 0;
  margin: 1em 0;
  font-size: 1rem;
  font-family: "Helvetica", "Segoe UI", Arial, sans-serif;
  font-weight: 400;
}

.NASA1975-article-author li {
  line-height: 1.2;
}

.NASA1975-article-author li::before {
  display: none;
}
