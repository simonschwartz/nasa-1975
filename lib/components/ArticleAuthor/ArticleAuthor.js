import React from "react";
import PropTypes from "prop-types";
import "./ArticleAuthor.css";

const ArticleAuthor = ({
  name,
  location = "",
  phone = "",
  className = "",
  ...attributeOptions
}) =>
  <ul className={`NASA1975-article-author ${className}`} {...attributeOptions}>
    <li>
      {name}
    </li>
    {location &&
      <li>
        {location}
      </li>}
    {phone &&
      <li>
        (Phone: {phone})
      </li>}
  </ul>;

ArticleAuthor.propTypes = {
  /** Name of article author */
  name: PropTypes.string.isRequired,
  /** Location of article author */
  location: PropTypes.string,
  /** Contact number for article author */
  phone: PropTypes.string,
  /** className to add to heading */
  className: PropTypes.string
};

export default ArticleAuthor;
