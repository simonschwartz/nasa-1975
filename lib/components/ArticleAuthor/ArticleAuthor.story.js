import React from "react";
import { text, select } from "@storybook/addon-knobs";
import { withInfo } from "@storybook/addon-info";
import ArticleAuthor from "./";

const ArticleAuthorStory = withInfo({
  text: "Display the author details for a news release"
})(() => {
  // Setup knob for setting author name text
  const labelName = "name";
  const defaultValueName = "John Smith";
  const nameValue = text(labelName, defaultValueName);

  // Setup knob for setting author location text
  const labelLocation = "location";
  const defaultValueLocation = "Goddard Space Flight Center";
  const locationValue = text(labelLocation, defaultValueLocation);

  // Setup knob for setting author phone text
  const labelPhone = "phone";
  const defaultValuePhone = "301/725-1234";
  const phoneValue = text(labelPhone, defaultValuePhone);

  return (
    <ArticleAuthor
      name={nameValue}
      location={locationValue}
      phone={phoneValue}
    />
  );
});

export default ArticleAuthorStory;
