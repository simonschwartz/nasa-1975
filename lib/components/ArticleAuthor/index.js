import ArticleAuthor from "./ArticleAuthor";

export default ArticleAuthor;
