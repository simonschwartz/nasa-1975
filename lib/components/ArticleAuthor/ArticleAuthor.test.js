import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ArticleAuthor from "./";

it("renders correct markup with no optional props", () => {
  const tree = renderer
    .create(<ArticleAuthor name="Test author name" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correct markup with location prop", () => {
  const tree = renderer
    .create(
      <ArticleAuthor
        name="Test author name"
        location="Goddard Space Flight Center"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correct markup with phone prop", () => {
  const tree = renderer
    .create(<ArticleAuthor name="Test author name" phone="301/725-1234" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correct markup with multiple props", () => {
  const tree = renderer
    .create(
      <ArticleAuthor
        name="Test author name"
        location="Goddard Space Flight Center"
        phone="301/725-1234"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ArticleAuthor name="Test author name" />, div);
});
