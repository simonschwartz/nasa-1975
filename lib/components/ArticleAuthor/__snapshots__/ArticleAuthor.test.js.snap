// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct markup with location prop 1`] = `
<ul
  className="NASA1975-article-author "
>
  <li>
    Test author name
  </li>
  <li>
    Goddard Space Flight Center
  </li>
  
</ul>
`;

exports[`renders correct markup with multiple props 1`] = `
<ul
  className="NASA1975-article-author "
>
  <li>
    Test author name
  </li>
  <li>
    Goddard Space Flight Center
  </li>
  <li>
    (Phone: 
    301/725-1234
    )
  </li>
</ul>
`;

exports[`renders correct markup with no optional props 1`] = `
<ul
  className="NASA1975-article-author "
>
  <li>
    Test author name
  </li>
  
  
</ul>
`;

exports[`renders correct markup with phone prop 1`] = `
<ul
  className="NASA1975-article-author "
>
  <li>
    Test author name
  </li>
  
  <li>
    (Phone: 
    301/725-1234
    )
  </li>
</ul>
`;
